### ISSUE TITLE
Out of bounds colors in new discrete normal map

### ISSUE THREAD

**Issue by behreajj** [aseprite/aseprite#3594]
The normal map looks much better after 20902e3 . There is an issue with colors going out of bounds in the discrete wheel though. For example, `#ffcc59` should be `#eebf80`. A proposed change to fix this would be

```hlsl
nd = max(0.0, (ceil(nd * 6.0) - 1.0) / 5.0);
```

in the shader

```cpp
normalizedDistance = (std::ceil(normalizedDistance * 6.0) - 1.0) / 5.0;
if (normalizedDistance < 0.0) { normalizedDistance = 0.0; }
```

in the cpu version (or use a `max` function).

Demos graph: https://www.desmos.com/calculator/t7u4hgcosb .

**Comment by dacap** [comment]
Thanks @behreajj, seeing the current values:

![Screenshot from 2022-11-17 15-34-08](https://user-images.githubusercontent.com/39654/202529389-c7597c0b-16cb-4f23-b24b-edf7a482433b.png)

and with your change:

![Screenshot from 2022-11-17 15-34-46](https://user-images.githubusercontent.com/39654/202529417-558c41e0-bbc1-4452-9e07-a733bc7b8591.png)

I think the current picture is correct, but the values returned by the eyedropper (when picking colors outside the wheel radius and in discrete mode) are incorrect (but the colors, if you take a screenshot of these wheels, the first one is correct, I think :thinking: )

I'll try to fix this.

**Comment by behreajj** [comment]
Yeah dunno. Looked like I checked with shaders turned off, and it looked like ok aside from the distribution (0, 36, 54... inclination).  Unfortunately, I can't post pictures to Github at the moment.

**Comment by behreajj** [comment]
After messing around with it some more, I'm suspecting that gamma correction be the issue, as in `pow(d.x, 2.2)`? (If so, that would unfortunately mean that non-standard color profiles would mess with the normal map appearance's even more.)

**Comment by dacap** [comment]
Interesting, didn't thought about that, anyway the RGB values when picking a color, are calculated with getMainAreaColor() (as used by ColorSelector::getColorByPosition()). And the color wheel is not using color profiles (maybe the appearance of the colors in the screen could change, but clicking a position in the wheel calculates the color, doesn't pick the screen color).
